x-pack/plugins/ml/public/application/components/data_grid/common.ts

```diff
@@ -82,6 +83,37 @@ export const getFieldsFromKibanaIndexPattern = (indexPattern: IndexPattern): str
   return indexPatternFields;
 };
 
+/**
+ * Return a map of runtime_mappings for each of the index pattern field provided
+ * to provide in ES search queries
+ * @param indexPatternFields
+ * @param indexPattern
+ * @param clonedRuntimeMappings
+ */
+export const getRuntimeFieldsMapping = (
+  indexPatternFields: string[],
+  indexPattern: IndexPattern,
+  clonedRuntimeMappings?: { [key: string]: RuntimeField }
+) => {
+  if (!Array.isArray(indexPatternFields) || indexPattern === undefined) return {};
```

Comment:
From the type definition, it seems like `indexPatternFields` should be an array, but then you check it with `isArray`. Should it be `indexPatternFields: string[] | undefined` instead? Same about `intexPatter`, `indexPattern: IndexPattern | undefined`

Reply:
Updated here https://github.com/elastic/kibana/pull/90363/commits/08e635ec2c6cd9fadf13d1b0002f4978b0103527